crates/rome_js_analyze/src/analyzers/no_unsafe_negation.rs

```diff
@@ -0,0 +1,156 @@
+use crate::JsRuleAction;
+use rome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Rule, RuleCategory, RuleDiagnostic,
+};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_factory::make;
+use rome_js_syntax::{JsAnyExpression, JsInExpression, JsInstanceofExpression, T};
+use rome_rowan::{declare_node_union, AstNode, AstNodeExt};
+
+declare_rule! {
+    /// Disallow using unsafe negation.
+    ///
+    /// ## Examples
+    ///
+    /// ### Valid
+    /// ```js
+    /// -1 in [1,2];
+    /// ```
+    /// ```js
+    /// ~1 in [1,2];
+    /// ```
+    /// ```js
+    /// typeof 1 in [1,2];
+    /// ```
+    /// ```js
+    /// void 1 in [1,2];
+    /// ```
+    /// ```js
+    /// delete 1 in [1,2];
+    /// ```
+    /// ```js
+    /// +1 instanceof [1,2];
+    /// ```
+    /// ### Invalid
+
+    /// ```js,expect_diagnostic
```

Comment:
```suggestion
    /// ### Invalid
    /// ```js,expect_diagnostic
```